skmultiflow/classification/trees/hoeffding_adaptive_tree.py

```diff
@@ -0,0 +1,489 @@
+                    # To check, bound never less than (old_error_rate - alt_error_rate)
+                    if bound < (old_error_rate - alt_error_rate):
+                        hat._active_leaf_node_cnt -= self.number_leaves()
+                        hat._active_leaf_node_cnt += self._alternate_tree.number_leaves()
+                        self.kill_tree_childs(hat)
+
+                        if parent is not None:
+                            parent.set_child(parent_branch, self._alternate_tree)
+                        else:
+                            hat._tree_root = hat._tree_root.alternateTree
+                        hat._switchAlternateTrees += 1
+                    elif (bound < alt_error_rate - old_error_rate):
+                        if isinstance(self._alternate_tree, HAT.ActiveLearningNode):
+                            self._alternate_tree = None
+                        elif (isinstance(self._alternate_tree, HAT.ActiveLearningNode)):
+                            self._alternate_tree = None
+                        else:
+                            self._alternate_tree.kill_tree_childs(hat)
+                        hat._prunedalternateTree += 1  # hat._pruned_alternate_trees to check
+
+            # Learn_From_Instance alternate Tree and Child nodes
+            if self._alternate_tree is not None:
+                self._alternate_tree.learn_from_instance(X, y, weight, hat, parent, parent_branch)
+
+            child_branch = self.instance_child_index(X)
+            child = self.get_child(child_branch)
+
+            if child is not None:
+                child.learn_from_instance(X, y, weight, hat, parent, parent_branch)
+
+        # Override NewNode
+        def kill_tree_childs(self, hat):
+            for child in self._children:
+                if child is not None:
+                    # Delete alternate tree if it exists
+                    # TODO (check the casting to object)
+                    # if (isinstance(child, HAT.AdaSplitNode) and HAT.NewNode(HAT.AdaSplitNode(child)) is not None):
+                    # HAT.NewNode(HAT.AdaSplitNode(child)).kill_Tree_Childs(hat)
+                    # self._pruned_alternate_trees += 1
+
+                    # Recursive delete of SplitNodes
+                    if isinstance(child, HAT.AdaSplitNode):
+                        child.kill_tree_childs(hat)
+
+                    if isinstance(child, HAT.ActiveLearningNode):
+                        child = None
+                        hat._active_leaf_node_cnt -= 1
+                    elif isinstance(child, HAT.InactiveLearningNode):
+                        child = None
+                        hat._inactive_leaf_node_cnt -= 1
+
+        # override NewNode
+        def filter_instance_to_leaves(self, X, parent, parent_branch, update_splitter_counts, found_nodes=None):
+            if found_nodes is None:
+                found_nodes = []
+            # TODO
```

Comment:
- [ ] Uncomment block.

```self._observed_class_distribution``` is a dictionary, the following section does not aplies:
``` python
if y > len(self._observed_class_distribution) - 1:
    return
```

Reply:
Done!

Reply:
Only the two lines mentioned above should be deleted.

- [ ] Get back the following code:
``` python
if y not in self._observed_class_distribution:
    self._observed_class_distribution[y] = 0.0
    self._observed_class_distribution[y] += weight # Dictionary (class_value, weight)
```